dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -169,27 +168,25 @@ def _check_dataset_time_consistency_by_dimensions(self, time_dim, load_data_df):
             .agg(F.countDistinct(*time_col).alias("distinct_timestamps"))
         )
 
-        distinct_counts = timestamps_by_dims.select("distinct_timestamps").distinct()
-        if distinct_counts.count() > 1:
+        distinct_counts = timestamps_by_dims.select("distinct_timestamps").distinct().collect()
```

Comment:
If this is taking too long for TEMPO, we can consider an alternative check. Instead of checking that each dim combo have the same timestamp count and that the count is expected, we can check that each timestamp has the same count (i.e.count df groupy timestamp). Maybe it would be faster considering TEMPO has many dim combos. And we already check in the time class func that timestamps match time range specified.

Reply:
I agree here. The smarter check being performed in the representative_time_config should be moved up here. This code can group by the set of timestamp columns instead of dimension combinations.

Reply:
FYI, Dimension combinations is just `id` col in the Standard handler b/c load_data is passed into the dataset time check. But yeah, we should just check count by timestamps.
https://github.com/dsgrid/dsgrid/blob/88529273139803da98482d56a51e3f4235cc0b36/dsgrid/dataset/dataset_schema_handler_standard.py#L41